src/variableProcessor.ts

```diff
@@ -215,30 +215,28 @@ export class VariableProcessor {
                                                 ? displayName.substring(0, displayNameSpaceIndex)
                                                 : displayName;
                                             let bestMatches = [];
+                                            const bestMatchesRegex = new RegExp(`(^${displayNameFirstWord}\.com$)|(^${displayNameFirstWord}\.[a-z]+(?:\.[a-z]+)?$)|(^${displayNameFirstWord}[a-z]+\.com$)|(^${displayNameFirstWord}[^:]*$)|(^[^:]*${displayNameFirstWord}[^:]*$)`, "gi");
+                                            const bestMatchesRegexGroups = (bestMatchesRegex.source.match(new RegExp(`${displayNameFirstWord}`, "g")) || []).length;

```

Comment:
Do we need this fallback?

Reply:
This is intended to protect us when the regex is updated to ensure we're always accounting for the correct number of capture groups. I can revert to the explicit 5 checks, but figured this would be more future-proof. I'm fine either way.

Reply:
@flanakin I agree with you that no hard code should exist, I just mean can we remove `|| []` like following:

```js
const bestMatchesRegexGroups = bestMatchesRegex.source.match(new RegExp(`${displayNameFirstWord}`, "g")).length
```

Reply:
Oh, yes! You're right. Since we know it'll match, we'll never have 0. Good catch!